vault/resource_database_secret_backend_connection.go

```diff
@@ -273,6 +273,59 @@ func databaseSecretBackendConnectionResource() *schema.Resource {
 				ConflictsWith: util.CalculateConflictsWith("oracle", dbBackendTypes),
 			},
 
+			"snowflake": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Connection parameters for the snowflake-database-plugin plugin.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The Account Name for the snowflake connection. Formats <accountname> if in us-west-2, or <accountname>.<region> for any other region",
+						},
```

Comment:
I'm curious why this schema breaks the `connection_url` API parameter up into its components? Is it just for convenience?

What do you think about just following the [API parameters](https://www.vaultproject.io/api/secret/databases/snowflake#parameters), and including good examples in the docs for different types of connection_url's? It also looks like in order to support root credential rotation, we need to be able to specify the `connection_url` separately from the `username` and `password`.

Also this schema seems to be missing the `max_open_connections`, `max_idle_connections`, `max_connection_lifetime`, and `username_template` parameters.